### PR TITLE
support gradient clipping for nnestimator

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

**Comment by hkvision** [comment]
Thanks. I will refer to your PR to modify the Keras API part.

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py [hunk 1]
```diff
@@ -271,6 +271,44 @@ def getValidation(self):
         """
         return self.validation_config
 
+    def disable_gradient_clipping(self):
```
[the remark this answers is not in the document]
Comment (dding3) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: Updated

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py [hunk 2]
```diff
@@ -271,6 +271,44 @@ def getValidation(self):
...
+        """
+        Call this if you would like to disable gradient clipping during the training process.
```
[the remark this answers is not in the document]
Comment (dding3) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: Removed note necessary comments.
Currently I didn't set default value for disableGradientClipping as I think it's needed after user called setclipping.

**Comment by hhbyyh** [comment]
Thanks for the update. Please correct me if I'm wrong, just to confirm my understanding:

By default, gradient clipping is not involved, regardless user invoke `disableGradientClipping` explicitly or not.

If user invoke `setConstantGradientClipping` or `setGradientClippingByL2Norm`, gradient clipping will be set to the optimizer.

It appears to me that disableGradientClipping is not necessary based on the understanding above.

**Comment by hhbyyh** [comment]
Synced with Ding, The function of disableGradientClipping can be replaced by
```
clear(l2GradientClippingParams) or
clear(constantGradientClippingParams)
```.
Intuitively, the order of invoking setters in Estimator should not have an impact on the fit result, to support grid search.

**Comment by dding3** [comment]
CC @hhbyyh

**Comment by hhbyyh** [comment]
LGTM

**Comment by dding3** [comment]
Thanks for your feedback. @hhbyyh 

Jekins pass. http://172.168.2.101:8080/job/ZOO-PR-Validation/349/

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala [hunk 3]
```diff
@@ -158,6 +170,22 @@ class NNEstimator[T: ClassTag] private[zoo] (
...
+  }
+
+  def setGradientClippingByL2Norm(clipNorm: Float): this.type = {
+    set(l2GradientClippingParams, clipNorm)
+  }
+
+  /**
+   * Clear clipping params, in this case, clipping will not be applied.
+   */
+  def clearGradientClippingParams(): this.type = {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: Change to `clearGradientClipping`
